client/src/partials/vouchers/records.js

```diff
@@ -0,0 +1,196 @@
+      },
+      { field : 'description', displayName : 'TABLE.COLUMNS.DESCRIPTION', headerCellFilter: 'translate',
+        groupingShowAggregationMenu: false
+      },
+      { field : 'amount', displayName : 'TABLE.COLUMNS.AMOUNT', headerCellFilter: 'translate',
+        treeAggregationType: uiGridGroupingConstants.aggregation.SUM,
+          groupingShowAggregationMenu: false
+      },
+      { field : 'user', displayName : 'TABLE.COLUMNS.RESPONSIBLE', headerCellFilter: 'translate',
+        groupingShowAggregationMenu: false
+      },
+      { field : 'action', displayName : '...',
+        cellTemplate: linkReceiptTemplate,
+        enableFiltering: false,
+        enableColumnMenu: false
+      }
+    ];
+
+  // register API
+  vm.gridOptions.onRegisterApi = onRegisterApi;
+
+  // expose function
+  vm.getType = getType;
+  vm.isDefined = isDefined;
+  vm.showReceipt = showReceipt;
+
+  // startup
+  startup();
+
+  // API register function
+  function onRegisterApi(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  // Grid Aggregation
+  function typeAggregation(aggregation) {
+    var type = getType(aggregation.groupVal);
+    aggregation.rendered = type ? $translate.instant(type.text) : $translate.instant('FORM.LABELS.UNDEFINED');
+  }
+
+  // Time Aggregation
+  function timeAggregation(aggregation) {
+    var date = new Date(aggregation.groupVal);
+    var time = date.getHours() + ':' + date.getMinutes();
+    aggregation.rendered = aggregation.groupVal ? date.toDateString().concat('  (', time, ')') : null;
+  }
+
+  // isDefined Type
+  function isDefined(row) {
+    return row.uuid && (row.origin_id === null || row.origin_id === undefined);
+  }
+
+  // get vouchers type
+  function getType(originId) {
+    if (originId === null || originId === undefined) { return {}; }
+
+    return Vouchers.transferType.filter(function (item) {
+      return item.id === originId;
+    })[0];
+  }
+
+  // enable filter
+  function toggleFilter() {
+    vm.filterEnabled = !vm.filterEnabled;
+    vm.bcButtons[0].color = vm.filterEnabled ? 'btn-default active' : 'btn-default';
+    vm.gridOptions.enableFiltering = vm.filterEnabled;
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.ALL);
+  }
+
+  // search voucher
+  function search() {
+    Modal.openDateInterval()
+    .then(function (dateInterval) {
+      vm.dateInterval = dateInterval;
+      return Vouchers.read(null, vm.dateInterval);
+    })
+    .then(function (list) {
+      // need only one line of the voucher transaction
+      vm.gridOptions.data = list.filter(function (item) {
+        return item.debit > 0;
+      });
+    })
+    .catch(Notify.errorHandler);
+  }
+
+  // print vouchers list
+  function printList() {
+    var dateFrom = vm.dateInterval ? util.htmlDate(vm.dateInterval.dateFrom) : null;
+    var dateTo = vm.dateInterval ? util.htmlDate(vm.dateInterval.dateTo) : null;
+    var url = '/vouchers/reports?dateFrom=' + dateFrom + '&dateTo=' + dateTo;
```

Comment:
You should really use `$http`'s parameter serializing.  Just do this:
```js
$http.get('/vouchers/reports', { params : { dateFrom : dateFrom, dateTo : dateTo }});
// this will produce the following query:
// GET /patients/search?dateFrom=2014-12-31T23:00:00.000Z&dateTo=2016-07-06T14:24:51.706Z
```

That way: 
 1. We are using a tested date serialization library, and do not depend on `util.htmlDate()`.
 2. We clearly separate the base URL from the parameters.
 3. We don't have to think about converting between types - angular will do it for us